### PR TITLE
support watch resource group

### PR DESCRIPTION
**Pull request by HuSharp**
Signed-off-by: husharp <jinhao.hu@pingcap.com>

### CODE REVIEW

**Review by disksing — APPROVED** [review]

**Comment by ekexium** [comment]
I think changes that break compatibility had better be merged together with dependent PRs. Otherwise it may block other changes

**Comment by HuSharp** [comment]
> I think changes that break compatibility had better be merged together with dependent PRs. Otherwise it may block other changes

Sry...My fault, should have made a draft label...
I'll do a tikv fix proto firstly

File: proto/pdpb.proto [hunk 1]
```diff
@@ -103,33 +103,43 @@ service PD {
...
-    repeated string names = 1;
+    string config_path = 1;
 }
 
 message LoadGlobalConfigResponse {
     repeated GlobalConfigItem items = 1;
+    int64 revision = 2;
+}
+
+enum ItemKind {
```
Comment (rleungx) [inline proto/pdpb.proto:133]: How about calling it EventType or something else which can represent that it is an action?

Comment (HuSharp) [reply proto/pdpb.proto:133]: You are right! from that way we can sync with Etcd

File: proto/pdpb.proto [hunk 2]
```diff
@@ -103,33 +103,43 @@ service PD {
...
 
 message WatchGlobalConfigResponse {
     repeated GlobalConfigItem changes = 1;
+    int64 revision = 2;
 }
 
 message StoreGlobalConfigRequest {
-    repeated GlobalConfigItem changes = 1;
+    string config_path = 1;
+    repeated GlobalConfigItem changes = 2;
```
Comment (Connor1996) [inline proto/pdpb.proto:117]: breaking change, shouldn't change the field number

Comment (HuSharp) [reply proto/pdpb.proto:117]: This interface was previously used by TopSQL as a configuration center control switch, but then chose another way.
tikv/pd/tidb are not use it.

**Pull request merged**